code/game/objects/structures/lavaland/ore_vent.dm

```diff
@@ -0,0 +1,509 @@
+		iterator++
+
+
+/**
+ * Returns the quantity of mineral sheets in each ore's boulder contents roll. First roll can produce the most ore, with subsequent rolls scaling lower logarithmically.
+ */
+/obj/structure/ore_vent/proc/ore_quantity_function(ore_floor)
+	var/mineral_count = boulder_size * (log(rand(1+ore_floor, 4+ore_floor))**-1)
+	mineral_count = SHEET_MATERIAL_AMOUNT * round(mineral_count)
+	return mineral_count
+
+/**
+ * Starts the wave defense event, which will spawn a number of lavaland mobs based on the size of the ore vent.
+ * Called after the vent has been tapped by a scanning device.
+ * Will summon a number of waves of mobs, ending in the vent being tapped after the final wave.
+ */
+/obj/structure/ore_vent/proc/start_wave_defense()
+	AddComponent(\
+		/datum/component/spawner, \
+		spawn_types = defending_mobs, \
+		spawn_time = (10 SECONDS + (5 SECONDS * (boulder_size/5))), \
+		max_spawned = 10, \
+		max_spawn_per_attempt = (1 + (boulder_size/5)), \
+		spawn_text = "emerges to assault", \
+		spawn_distance = 4, \
+		spawn_distance_exclude = 3, \
+	)
+	var/wave_timer = 60 SECONDS
+	if(boulder_size == BOULDER_SIZE_MEDIUM)
+		wave_timer = 90 SECONDS
+	else if(boulder_size == BOULDER_SIZE_LARGE)
+		wave_timer = 150 SECONDS
+	COOLDOWN_START(src, wave_cooldown, wave_timer)
+	addtimer(CALLBACK(src, PROC_REF(handle_wave_conclusion)), wave_timer)
+	spawning_mobs = TRUE
+	icon_state = icon_state_tapped
+	update_appearance(UPDATE_ICON_STATE)
+
+/**
+ * Called when the wave defense event ends, after a variable amount of time in start_wave_defense.
+ *
+ * If the node drone is still alive, the ore vent is tapped and the ore vent will begin generating boulders.
+ * If the node drone is dead, the ore vent is not tapped and the wave defense can be reattempted.
+ *
+ * Also gives xp and mining points to all nearby miners in equal measure.
+ */
+/obj/structure/ore_vent/proc/handle_wave_conclusion()
+	SEND_SIGNAL(src, COMSIG_VENT_WAVE_CONCLUDED)
+	COOLDOWN_RESET(src, wave_cooldown)
+	particles = null
+	if(!QDELETED(node)) ///The Node Drone has survived the wave defense, and the ore vent is tapped.
+		tapped = TRUE
+		SSore_generation.processed_vents += src
+		balloon_alert_to_viewers("vent tapped!")
+	else
+		visible_message(span_danger("\the [src] creaks and groans as the mining attempt fails, and the vent closes back up."))
+		icon_state = initial(icon_state)
+		update_appearance(UPDATE_ICON_STATE)
+		return FALSE //Bad end, try again.
+
+	for(var/mob/living/carbon/human/potential_miner as anything in range(7, src)) //Give the miners who are near the vent points and xp.
+		if(ishuman(potential_miner))
+			var/mob/living/carbon/human/true_miner = potential_miner
+			var/obj/item/card/id/user_id_card = true_miner.get_idcard(TRUE)
+			if(true_miner && true_miner.mind)
+				true_miner.mind.adjust_experience(/datum/skill/mining, MINING_SKILL_BOULDER_SIZE_XP * boulder_size)
+			if(!user_id_card)
+				continue
+			if(user_id_card)
```

Comment:
remove this if check and move code inside it out of its indentation. You already checked for this just 1 line above